baybe/utils/dataframe.py

```diff
@@ -413,3 +414,42 @@ def fuzzy_row_match(
             inds_matched.extend(inds_found)
 
     return pd.Index(inds_matched)
+
+
+def pretty_printing_dataFrame(df: pd.DataFrame) -> str:
+    """Return a pretyt/readable str Object.
+
+    Args:
+        df: The dataframe to be printed.
+
+    Returns:
+        The values to be printed as a str table.
+    """
+    # select the first 10 rows/values and store them in a str table
+    # TODO: remove hard coded numbers and allow user defined specifications
+
+    str_df = ""
+
+    # Handling with Empty dataFrame
+    if len(df) == 0:
+        return "This attribute is empty. "
+
+    # Extract extra columns
+    if len(df.columns) > 5:
+        df = df.T.head(5).T
+        str_df = (
+            "\n Your dataFrame has too many columns. "
+            + "For a better readable table only the first 5 columns will be printed. \n"
+        )
+
+    # Extract extra rows
+    if len(df) > 10:
+        df = df.head(10)
```

Comment:
Does this change the provided dataframe itself? I honestly do not know (I am always confused by call to value vs. call to reference stuff), but it feels like this could be the case here. Just to be sure, you can simply use a different variable name here and cleverly have it assigned with an in-line if-statement à la `printed_def = df.head(10) if len(df)>10 else df`.
Also, please check if something like this is also present in other places.

Reply:
Thank you Alex, for bringing this into discussion! I can change the variable name to avoid any confusion 👍